static/scripts/rewards/web3/erc20-permit.ts

```diff
@@ -182,3 +197,37 @@ export function nonceBitmap(nonce: BigNumberish): { wordPos: BigNumber; bitPos:
   const bitPos = BigNumber.from(nonce).and(255).toNumber();
   return { wordPos, bitPos };
 }
+
+export async function doesPermitHaveTxHash(permit: Erc20Permit): Promise<string | null> {
+  const { data, error } = await supabase
+    .from("permits")
+    .select("transaction")
+    // using only nonce in the condition as it's defined unique on db
+    .eq("nonce", permit.permit.nonce.toString());
+
+  if (data?.length == 1 && data[0].transaction !== null) {
+    return data[0].transaction;
+  }
+
+  if (error !== null) {
+    console.error(error);
+    throw error;
+  }
+
+  return null;
+}
+
+export async function updatePermitTxHash(permit: Erc20Permit, hash: string): Promise<void> {
+  const { error } = await supabase
+    .from("permits")
+    .update({ transaction: hash })
```

Comment:
We can't allow write access to anonymous users. There must be a mechanic where https://supabase.com/docs/guides/auth/column-level-security is set to allow writes only to authorized users (wallets) with a matching `permits.beneficiary_id`. Otherwise anybody can simply spoil the `permits` table.

Reply:
So in conclusion should we change settings in the database (can only update if `null`) and merge this code?

Reply:
sounds good to me. should I do the updates (thinking of using `before update` trigger functions) on the db?
I don't know if I have permissions though.

Reply:
> sounds good to me. should I do the updates (thinking of using `before update` trigger functions) on the db?
> 
> I don't know if I have permissions though. 

If I added you as a dev on the Supabase project then I believe you do!

Reply:
added related function to the supabase and tested on my local.